### PR TITLE
Feat/add redirect location to ovh_iploadbalancing_http_frontend

### PR DESCRIPTION
**Pull request by DamienJ-wiza**
[Issue](https://github.com/terraform-providers/terraform-provider-ovh/issues/131)

### CODE REVIEW

File: ovh/types_iploadbalancing.go [hunk 1]
```diff
@@ -119,16 +119,17 @@ type IpLoadbalancingTcpFrontend struct {
...
+	FrontendId       int      `json:"frontendId,omitempty"`
+	Port             string   `json:"port"`
+	Zone             string   `json:"zone"`
+	AllowedSource    []string `json:"allowedSource,omitempty"`
+	DedicatedIpFo    []string `json:"dedicatedIpfo,omitempty"`
+	DefaultFarmId    *int     `json:"defaultFarmId,omitempty"`
+	DefaultSslId     *int     `json:"defaultSslId,omitempty"`
+	Disabled         *bool    `json:"disabled"`
+	Ssl              *bool    `json:"ssl"`
+	RedirectLocation string   `json:"redirectLocation,omitempty"`
```
Comment (yanndegat) [inline ovh/types_iploadbalancing.go]: i think should be *string as the attribute is optional

File: ovh/resource_ovh_iploadbalancing_http_frontend.go [hunk 2]
```diff
@@ -153,13 +159,14 @@ func resourceIpLoadbalancingHttpFrontendUpdate(d *schema.ResourceData, meta inte
...
-		Disabled:      getNilBoolPointerFromData(d, "disabled"),
-		Ssl:           getNilBoolPointerFromData(d, "ssl"),
-		DisplayName:   d.Get("display_name").(string),
+		Port:             d.Get("port").(string),
+		Zone:             d.Get("zone").(string),
+		AllowedSource:    allowedSources,
+		DedicatedIpFo:    dedicatedIpFo,
+		Disabled:         getNilBoolPointerFromData(d, "disabled"),
+		Ssl:              getNilBoolPointerFromData(d, "ssl"),
+		RedirectLocation: d.Get("redirect_location").(string),
```
Comment (yanndegat) [inline ovh/resource_ovh_iploadbalancing_http_frontend.go]: if we switch to *string, then we should use getNilStringPointerFromData(d, "redirect_localtion")

File: ovh/resource_ovh_iploadbalancing_http_frontend.go [hunk 3]
```diff
@@ -95,13 +100,14 @@ func resourceIpLoadbalancingHttpFrontendCreate(d *schema.ResourceData, meta inte
...
```
Comment (yanndegat) [inline ovh/resource_ovh_iploadbalancing_http_frontend.go]: if we switch to *string, then we should use getNilStringPointerFromData(d, "redirect_localtion")

File: ovh/resource_ovh_iploadbalancing_http_frontend.go [hunk 4]
```diff
@@ -176,6 +183,7 @@ func readIpLoadbalancingHttpFrontend(r *IpLoadbalancingHttpFrontend, d *schema.R
 	d.Set("display_name", r.DisplayName)
 	d.Set("port", r.Port)
 	d.Set("zone", r.Zone)
+	d.Set("redirect_location", r.RedirectLocation)
```
Comment (yanndegat) [inline ovh/resource_ovh_iploadbalancing_http_frontend.go]: as *string: should be 
if r.RedirectLocation != nil {
d.Set("redirect_location", *r.RedirectLocation)
}

File: website/docs/r/iploadbalancing_http_frontend.markdown [hunk 5]
```diff
@@ -50,7 +50,7 @@ The following arguments are supported:
 * `default_ssl_id` - Default ssl served to your customer
 * `disabled` - Disable your frontend. Default: 'false'
 * `ssl` - SSL deciphering. Default: 'false'
-
+* `redirect_location` - Redirection HTTP'
```
Comment (yanndegat) [inline website/docs/r/iploadbalancing_http_frontend.markdown]: usually, i try to "copy/paste" the official api field description
( but im not always consistent on this)

"HTTP redirection (Ex : http://www.ovh.com)"

File: ovh/resource_ovh_iploadbalancing_http_frontend.go [hunk 6]
```diff
@@ -110,13 +115,14 @@ func resourceIpLoadbalancingHttpFrontendCreate(d *schema.ResourceData, meta inte
...
-		Disabled:      d.Get("disabled").(bool),
-		Ssl:           d.Get("ssl").(bool),
-		DisplayName:   d.Get("display_name").(string),
+		Port:             d.Get("port").(string),
+		Zone:             d.Get("zone").(string),
+		AllowedSource:    allowedSources,
+		DedicatedIpFo:    dedicatedIpFo,
+		Disabled:         d.Get("disabled").(bool),
+		Ssl:              d.Get("ssl").(bool),
+    RedirectLocation: d.Get("redirect_location").(string),
```
Comment (pgaxatte) [inline ovh/resource_ovh_iploadbalancing_http_frontend.go:124]: This indentation has shifted

File: ovh/resource_ovh_iploadbalancing_http_frontend.go [hunk 7]
```diff
@@ -189,13 +197,14 @@ func resourceIpLoadbalancingHttpFrontendUpdate(d *schema.ResourceData, meta inte
...
-		AllowedSource: allowedSources,
-		DedicatedIpFo: dedicatedIpFo,
-		Disabled:      d.Get("disabled").(bool),
-		Ssl:           d.Get("ssl").(bool),
-		DisplayName:   d.Get("display_name").(string),
+		Port:             d.Get("port").(string),
+		Zone:             d.Get("zone").(string),
+		AllowedSource:    allowedSources,
+		DedicatedIpFo:    dedicatedIpFo,
+    Disabled:         d.Get("disabled").(bool),
```
Comment (pgaxatte) [inline ovh/resource_ovh_iploadbalancing_http_frontend.go:204]: This indentation has shifted

**Review by pgaxatte — CHANGES_REQUESTED** [review]
Discussed with @yanndegat maybe we don't need to change the `string` to `*string` on RedirectLocation and we can add a test later on to see if it is needed or not.

The two indentation problems have likely been introduced by your conflict resolution @yanndegat

**Review by yanndegat — APPROVED** [review]

**Review by pgaxatte — APPROVED** [review]
All good now, lgtm

**Pull request merged**